### DISCUSSION TITLE
Why isn’t it recommended to use position:absolute?

### DISCUSSION THREAD

**Discussion by ghost** [community/community#67845] [Programming Help]
### Body

Hi,
 
 I am coding an app in position absolute and I was wondering why my div would move from the bottom to more to the top when it minimize the screen and how to fix it?

### Guidelines

- [X] I have read and understood this category's [guidelines](https://github.com/orgs/community/discussions/51384) before making this post.

**Comment by leanczo** [comment]
Using position: absolute; isn't inherently bad, but it can create challenges in layout and design, especially when designing responsive web applications or websites. Here's why:

**Context Dependency**: An element with position: absolute; is positioned relative to its nearest positioned ancestor (if any); if none exists, it's positioned relative to the initial containing block. This means that if you aren't careful, the absolute positioning might rely on unintended parent elements, causing unexpected behavior.

**Document Flow**: Absolute positioning removes an element from the normal document flow. This means other elements will behave as though the absolutely positioned element doesn't exist, potentially causing overlapping and other layout issues.

**Responsiveness**: As you noticed, absolute positioning can lead to issues on different screen sizes or when the window is resized. When you define positions in terms of pixels (or even sometimes percentages), the design may not adapt well to varying screen dimensions or resolutions.

To address your specific problem of the div moving when you minimize the screen:

**Parent Context**: Ensure that the parent container of the div has position: relative;. This way, the div's position will be based on the parent container, and not some other unintended element or the entire viewport.

**Units**: Instead of using pixel values, consider using viewport units (vh, vw) or percentages, as they can be more flexible in response to screen resizing.

**Media Queries**: Use media queries to adjust styles based on the screen width or other properties. This allows you to provide different styles for different device or viewport sizes, enhancing the responsiveness of your design.

**Consider Flexbox or Grid**: For more complex layouts, consider using CSS Flexbox or Grid. They offer more flexibility and control for responsive design without the need for absolute positioning.

**Examples:**
**Flexbox**: With Flexbox, you can easily center elements on both axes.
```
.container {
    display: flex;
    justify-content: center; /* Center horizontally */
    align-items: center; /* Center vertically */
    height: 100vh; /* Ensure the container has a height */
}
```

**Grid**: CSS Grid is another powerful layout tool. You can use it to center items.
```
.container {
    display: grid;
    place-items: center; /* Center on both axes */
    height: 100vh; /* Ensure the container has a height */
}

```
**Transformation**: This technique involves using position: absolute; together with transformation properties.
```
.element {
    position: absolute;
    top: 50%;
    left: 50%;
    transform: translate(-50%, -50%);
}

```
**Margin Auto**: In certain contexts, especially when the element has a fixed width, you can use margin: auto.
```
.element {
    width: 300px;
    height: 300px;
    margin: auto;
    position: absolute;
    top: 0;
    bottom: 0;
    left: 0;
    right: 0;
}

```
_This is written with chatgpt_
